test/product-test.js

```diff
@@ -0,0 +1,299 @@
+
+// Sample produts
+const parsedJSON = require('../mongo-seed/product.json');
+
+describe('Products', () => {
+
+  // Clean up db
+  beforeEach((done) => {
+    mongoose.connect(process.env.MONGO_DB_URI || 'mongodb://localhost/btrshop-cloud', () => {
+      mongoose.connection.db.dropDatabase(() => {
+        // console.log('db drop') ;
+        mongoose.connection.collection('products').insertMany(parsedJSON, (err, r) => {
+          // console.log('db feeded');
+          done();
+        });
+      });
+
+    });
+  });
+
+  describe('Find', () => {
+    it('should return list of product', (done) => {
+      server
+      .get('/products')
+      .expect('Content-type', /json/)
+      .expect(200)
+      .end((err, res) => {
+        res.status.should.equal(200);
+        res.body.length.should.be.eql(3);
+        done();
+      });
+    });
+
+    it('should return 400 cause of bad param', (done) => {
+      server
+      .get('/products?test=0885909462872')
+      .expect('Content-type', /json/)
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return 400 cause of empty param', (done) => {
+      server
+      .get('/products?test=')
+      .expect('Content-type', /json/)
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return a 400 cause of string ean', (done) => {
+      server
+      .get('/products?ean=dfdfdffd')
+      .expect('Content-type', /json/)
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return a 400 cause of too short ean', (done) => {
+      server
+      .get('/products?ean=123')
+      .expect('Content-type', /json/)
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return a 400 cause of too long ean', (done) => {
+      server
+      .get('/products?ean=123456789123456789')
+      .expect('Content-type', /json/)
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+
+    it('should return a array with one single product', (done) => {
+      server
+      .get('/products/?ean=0885909462872')
+      .expect('Content-type', /json/)
+      .expect(200)
+      .end((err, res) => {
+        res.status.should.equal(200);
+        res.body.length.should.be.eql(1);
+        done();
+      });
+    });
+
+    it('should return a single product', (done) => {
+      server
+      .get('/products/0885909462872')
+      .expect('Content-type', /json/)
+      .expect(200)
+      .end((err, res) => {
+        res.status.should.equal(200);
+        res.body.ean.should.equal('0885909462872');
+        done();
+      });
+    });
+
+    it('should return an error because product dont exist', (done) => {
```

Comment:
This test is strange, it seems you are testing the returned HTTP code both, with 2 different values.
To be confirmed. Check others if needed.